### PR TITLE
add(jest): unit tests for Hounddog and Logger

### PR DESCRIPTION
**Pull request by drepram**
**What this PR does** 📖

Add unit tests (100% coverage) for the Hounddog and Logger utilities.

**Which issue(s) this PR fixes** 🔨

AP-531

**Special notes for reviewers** 🗒️

I've added a comment on JIRA at this ticket regarding how we can bump up the branches coverage of Hounddog.test.ts.

**Additional comments** 🎤

-

### CODE REVIEW

File: utilities/Hounddog.test.ts [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+      //         reactions: {
+      //             ["reaction1"]: {}
+      //         },
+      //         lastInbound: 123, // the last time a message was received by any member of conversation, other than account owner
+      //         lastUpdate: 123, // the last time a message was received by any member of conversation, including account owner
+      //         limit: 123,
+      //         skip: 123,
+      //         end: false,
+      //     }
+      //     },
```
Comment (stavares843) [inline utilities/Hounddog.test.ts:126]: please remove the commented out 🔨

File: utilities/Hounddog.test.ts [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+        },
+        textilePubkey: "https://accounts.google.com/o/oauth2/revoke?token=%s",
+        status: "",
+        state: "idle",
+        unreadCount: 123,
+        profilePicture: "",
+        badge: "community",
+        userAccount: "",
+        mailboxId: "",
+        // textilePubkey: "", this have been written above. cannot have redundancies.
```
Comment (stavares843) [inline utilities/Hounddog.test.ts:318]: please remove the commented out 🔨

File: utilities/Hounddog.test.ts (hunk 1 above)
Comment (drepram) [reply utilities/Hounddog.test.ts]: Updated. Please check!

File: utilities/Hounddog.test.ts (hunk 2 above)
Comment (drepram) [reply utilities/Hounddog.test.ts]: Updated. Please check!

File: utilities/Hounddog.test.ts [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+      status: 429,
+      fromMailboxId: "12345",
+      toMailboxId: "v4.0.0-rc.4",
+      to: "./path/to/file"
+    },
+    textilePubkey: "https://accounts.google.com/o/oauth2/revoke?token=%s",
+  }
+
+  beforeEach(() => {
+    // WebRTC.default.setActiveCall(state.webrtc, '0xdf9eb223bafbe5c5271415c75aecd68c21fe3d7e')
```
Comment (stavares843) [inline utilities/Hounddog.test.ts:317]: commented out code 🔨

File: utilities/Hounddog.test.ts [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+  test("Return the first retrieved active call", () => { // matchesActiveCall => state.friends.all.find (ref. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find)
+    let result: any = inst.matchesActiveCall(state)
+    expect(result).toMatchSnapshot()
+  })
+
+  test("Returns true if at least one active call is retrieved", () => { // matchesSomeActiveCall => state.friends.all.some (ref. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)
+    let result: any = inst.matchesSomeActiveCall(state)
+    expect(result).toMatchSnapshot()
+  })
+})
```
Comment (stavares843) [inline utilities/Hounddog.test.ts:333]: missing empty line before describe

File: utilities/Hounddog.test.ts (hunk 3 above)
Comment (drepram) [reply utilities/Hounddog.test.ts]: Removed. Please check again, thanks!

File: utilities/Hounddog.test.ts (hunk 4 above)
Comment (drepram) [reply utilities/Hounddog.test.ts:332]: Added. Please check again, thanks!

**Comment by drepram** [comment]
Hi, `git submodule update` wasn't reverting the changes so I went into the directory and checkout into the base commit for it (https://github.com/Satellite-im/locales/compare/53d940e23eb9ccfad2834aeab60fd19e399f36d6...1bd50dfae3a32cb15b6adb3884cd1d77921f3311#diff-663ade211b3a1552162de21c4031fcd16be99407aae5ceecbb491a2efc43d5d2); and that seemed to work, let me know if I am doing it wrong!

File: utilities/Hounddog.ts [hunk 5]
```diff
@@ -38,32 +48,49 @@ export default class Hounddog {
     })
   }
 
-  findFriendByAddress(
```
Comment (WanderingHogan) [inline utilities/Hounddog.ts:41]: please put this back

File: utilities/Hounddog.ts [hunk 6]
```diff
@@ -1,17 +1,20 @@
 import Vue from 'vue'
+import { FriendStatus } from '~/libraries/Solana/FriendsProgram/FriendsProgram.types'
```
Comment (WanderingHogan) [inline utilities/Hounddog.ts:2]: why did you add this? it doesn't look like it gets called?

**Review by WanderingHogan — CHANGES_REQUESTED** [review]
couple changes in the hounddog file

File: utilities/Hounddog.ts (hunk 6 above)
Comment (drepram) [reply utilities/Hounddog.ts]: Removed!

File: utilities/Hounddog.ts (hunk 5 above)
Comment (drepram) [reply utilities/Hounddog.ts:41]: I've put it back in and wrote an additional unit test for it. Please check!

Comment (WanderingHogan) [reply utilities/Hounddog.ts:41]: sweet, thank you

**Review by WanderingHogan — APPROVED** [review]

**Pull request merged**